#! /usr/bin/env python
# safety_controller reaction configuration

PACKAGE='ls01b_v2'

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("special_range",           double_t, 0,
           "Special range value for truncated laser data",
            0.0, -100.0, 100)
gen.add("robot_radius",           double_t, 0,
           "Robot's radius used for truncated laser data",
           1.0, -100.0, 100.0)
gen.add("center_x",           double_t, 0,
            "laser frame x-position related to Base",
            0.0, -5.0, 5.0)
gen.add("center_y",           double_t, 0,
            "laser frame x-position related to Base",
            0.0, -5.0, 5.0)
gen.add("truncated_mode",     int_t, 0,
            "Different approach to truncate laser data",
            0, 0, 4)
gen.add("angle_compensate",     bool_t, 0,
        "Angle compensation for raw laser data",
        True, False, True)

exit(gen.generate(PACKAGE, "safety_dynamic_param", "Filter"))